pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,170 @@
+	}
+}
+
+func secretsEncryptOps(action, product, cpIp string, ips []string) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
+		time.Sleep(20 * time.Second) // Wait for reencrypt action to complete before restarting services
+	}
+	for i, node := range ips {
+		nodearr := []string{node}
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
+		Expect(errRestart).NotTo(HaveOccurred(), "error restart service for node: "+nodeIp)
+		// Order of reboot matters. Etcd first then control plane nodes.
+		// Little lag needed between node restarts to avoid issues.
+		waitEtcdErr := shared.WaitForPodsRunning(5, 4, false)
+		if waitEtcdErr != nil {
+			shared.LogLevel("WARN", "pods not up after 20 seconds.")
+			if i != len(ips) {
+				shared.LogLevel("DEBUG", "continue service restarts")
+			}
+		}
+	}
+	switch product {
+	case "k3s":
+		waitPodsErr := shared.WaitForPodsRunning(5, 6, false)
+		if waitPodsErr != nil {
+			shared.LogLevel("WARN", "pods not up after 30 seconds")
+		}
+	case "rke2":
+		waitPodsErr := shared.WaitForPodsRunning(5, 12, false)
+		if waitPodsErr != nil {
+			shared.LogLevel("WARN", "pods not up after 60 seconds")
+		}
+	}
+
+	secretEncryptStatus, errGetStatus := waitForHashMatch(cpIp, product, 5, 36) // Max 3 minute wait time for hash to match
+	Expect(errGetStatus).NotTo(HaveOccurred(), "error getting secret-encryption status")
+	verifyStatusOutput(action, secretEncryptStatus)
+
+	errLog := logEncryptionFileContents(ips, product)
+	Expect(errLog).NotTo(HaveOccurred())
+}
+
+func waitForHashMatch(cpIp, product string, defaultTime time.Duration, times int) (string, error) {
+	var secretEncryptStatus string
+	var errGetStatus error
+	for i := 1; i <= times; i++ {
+		secretEncryptStatus, errGetStatus = shared.SecretEncryptOps("status", cpIp, product)
+		if errGetStatus != nil {
+			shared.LogLevel("DEBUG", "error getting secret-encryption status. Retry.")
+		}
+		if secretEncryptStatus != "" && strings.Contains(secretEncryptStatus, "All hashes match") {
+			shared.LogLevel("DEBUG", "Total sleep time before hashes matched: %d seconds", i*int(defaultTime))
+			return secretEncryptStatus, nil
+		}
+		time.Sleep(defaultTime * time.Second)
+	}
+	shared.LogLevel("WARN", "Hashes did not match after %d seconds", times*int(defaultTime))
+	return secretEncryptStatus, errGetStatus
+}
+
+func verifyStdOut(action, stdout string) {
+	switch action {
+	case "prepare":
```

Comment:
so here in `verifyStdOut` and `verifyStatusOutput` are the same "actions"  right, could we think in a better design for this ? 
idk but maybe i structure that holds action or the messages 
map them and check all if exists and if contains what u want  in one single place ? 

this rn is too long and repetitive unnecessary 
if you need ideas let me know i can also help

Reply:
No, they are not. 
verifyStdOut is the 'std output' before reboots for different actions - the outputs of secrets-encrypt prepare/rotate etc. 
Then we perform reboot services.  
The verifyStatusOutput is after - when we get the "secrets-encryption status". They have different contents/expectations.

Reply:
Sample Outputs: 
verifyStdOut verifies: 
```
$ sudo k3s secrets-encrypt rotate 
 rotate completed successfully
```

verifyStatusOutput verifies: 
```
$ sudo k3s secrets-encrypt status 
 Encryption Status: Enabled
Current Rotation Stage: rotate
Server Encryption Hashes: All hashes match

Active  Key Type  Name
------  --------  ----
 *      AES-CBC   aescbckey-2024-04-04T21:30:30Z
        AES-CBC   aescbckey
```

Reply:
In verifyStatusOutput, moved 
```
	Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
	Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
```
to outside the switch case. They are common for all actions.

Reply:
Also renamed the methods to avoid confusion and added method comments. 
verifyStdOut -> verifyActionStdOut
verifyStatusOutput -> verifyStatusStdOut

Reply:
> No, they are not. verifyStdOut is the 'std output' before reboots for different actions - the outputs of secrets-encrypt prepare/rotate etc. Then we perform reboot services. The verifyStatusOutput is after - when we get the "secrets-encryption status". They have different contents/expectations.


sorry .... my bad, i was not clear on my comment

i was saying that both functions has the same actions to perform on it , regardless the purpose or whatever ....then based on that my suggestion to improve code design to avoid over usage os switch statement which can easily lead to code duplication, making maintenance harder sometimes. 

so here would be really fit some table driven approach as simpler method, but could also use something like strategy pattern. 

this is only a recommendation, since its a test case file not an implementation of nothing shareable or etc

Reply:
i feel like thats overkill for the purpose here, plus they r methods no one else will re-use - ever - it belongs within testfile like you mentioned. Will keep this suggestion in mind for the future. Thanks.

Reply:
no p at all